source/sync_order.rst

```diff
@@ -0,0 +1,456 @@
+    
+    {"detail": "Invalid token"}
+
+If HTTP is used instead of HTTPS:
+
+.. code-block:: bash
+
+    HTTP/1.0 403 FORBIDDEN
+    Vary: Accept
+    Content-Type: application/json
+    Allow: POST, OPTIONS
+    [Redacted Header]
+
+    {"detail": "Please use https instead of http"}
+
+
+3. Sync Order
+====================
+| URL endpoint: https://host.com/api/store/orders/sync
+| Allowed method: GET
+| Requires authentication: Yes
+
+A. Request
+----------
+
+You sync orders which have not been synced by calling the API with these parameters
+
+Header
+______
+
+=================== =========== =======================
+Parameter           Required    Description
+=================== =========== =======================
+Content-Type        Yes         application/json
+Authorization       Yes         store token string
+=================== =========== =======================
+
+.. code-block::
+
+    Content-Type: application/json
+    Authorization: token vE53k50FVtct50ll8iHBE6FgMRVCyJeF
+
+Body
+____
+
+=================== =========== =======================
+Parameter           Required    Description
+=================== =========== =======================
+store               Yes         Store Code string
+=================== =========== =======================
+
+Here's an example of how the API call might look like in JSON format:
+
+.. code-block::
+
+    {
+        "store": "Store1"
+    }
+
+
+Example of API call request using cURL (JSON). To avoid HTTP 100 Continue, please specify "Expect:" as a header.
+
+.. code-block:: bash
+
+    $ curl -X POST -H "Content-Type: application/json" -H "Authorization: token vE53k50FVtct50ll8iHBE6FgMRVCyJeF" -H "Expect:" https://host.com/api/store/orders/sync -i -d '{ "store": "Store1" }'
+    
+
+B. Response
+-----------
+
+In response to these API calls, Omni will reply with the following data in JSON:
+
+=================== ==================
+Variable            Description
+=================== ==================
+orders              An array of order objects
+=================== ==================
+
+Omni replies with an array of up to 100 order objects which have not been synced, and each order object has the following `fields`_.
+
+If all orders are synced, it returns an empty array :code:`{ "orders" : [] }`
+
+
+Here are examples of API responses:
+
+
+If call to order status API is successful (JSON):
+
+.. code-block:: bash
+
+    HTTP/1.0 200 OK
+    Vary: Accept
+    Content-Type: application/json
+    Allow: POST, OPTIONS
+    [Redacted Header]
+
+    {
+        "orders": [
+            {
+                [Redacted Content]
+            },
+            {
+                [Redacted Content]
+            },
+            {
+                [Redacted Content]
+            }
+        ]
+    }
+
+When some fields don't validate:
+
+**NOTE**: Even if the store field within the request body contains an "invalid" value, it would still return an empty array :code:`{ "orders" : [] }`
```

Comment:
oh tadi commentnya ilang, @martinusindra tolong verify ini intended behaviour dari code kita ya

Reply:
<img width="967" alt="Screen Shot 2019-07-31 at 11 24 39 AM" src="https://user-images.githubusercontent.com/52644069/62183838-dfe72300-b385-11e9-94b6-02979afac506.png">

Reply:
@riv10dean Thanks, I have no doubt that's how it currently works.

What I'm trying to verify is that is this a bug or an intentional feature from us?

Reply:
emang begini ko, klo udah pop sekali, request berikutnya ordernya `[]`

Reply:
ok

Reply:
This is a `get` API, it should not have any body request.
and this API doesn't need any parameter, since it will get the store from the token which it pass from the header.

Reply:
misunderstood, if store_code is invalid, it should return invalid store code

Reply:
OK @riv10dean, as @martinusindra said, this API shouldn't have any request body. So this part needs to be deleted

Reply:
ok